pallets/funding/src/settlement.rs

```diff
@@ -0,0 +1,390 @@
+		ensure!(matches!(project_details.status, ProjectStatus::FundingFailed), Error::<T>::NotAllowed);
+
+		// Check if the bidder has a future deposit held
+		let contributor = contribution.contributor;
+
+		// Return the funding assets to the contributor
+		Self::release_funding_asset(
+			project_id,
+			&contributor,
+			contribution.funding_asset_amount,
+			contribution.funding_asset,
+		)?;
+
+		// Release the held PLMC bond
+		Self::release_bond(project_id, &contributor, contribution.plmc_bond)?;
+
+		// Remove the bid from the storage
+		Contributions::<T>::remove((project_id, contributor.clone(), contribution.id));
+
+		Self::deposit_event(Event::ContributionSettled {
+			project_id,
+			account: contributor,
+			id: contribution.id,
+			ct_amount: Zero::zero(),
+		});
+
+		Ok(())
+	}
+
+	fn mint_ct_tokens(project_id: ProjectId, participant: &AccountIdOf<T>, amount: BalanceOf<T>) -> DispatchResult {
+		if !T::ContributionTokenCurrency::contains(&project_id, participant) {
+			T::ContributionTokenCurrency::touch(project_id, participant, participant)?;
+		}
+		T::ContributionTokenCurrency::mint_into(project_id, participant, amount)?;
+		Ok(())
+	}
+
+	fn release_funding_asset(
+		project_id: ProjectId,
+		participant: &AccountIdOf<T>,
+		amount: BalanceOf<T>,
+		asset: AcceptedFundingAsset,
+	) -> DispatchResult {
+		let project_pot = Self::fund_account_id(project_id);
+		T::FundingCurrency::transfer(
+			asset.to_assethub_id(),
+			&project_pot,
+			&participant,
+			amount,
+			Preservation::Expendable,
+		)?;
+		Ok(())
+	}
+
+	fn release_bond(project_id: ProjectId, participant: &AccountIdOf<T>, amount: BalanceOf<T>) -> DispatchResult {
+		// Release the held PLMC bond
+		T::NativeCurrency::release(
+			&HoldReason::Participation(project_id).into(),
+			&participant,
+			amount,
+			Precision::Exact,
+		)?;
+		Ok(())
+	}
+
+	fn slash_evaluator(project_id: ProjectId, evaluation: &EvaluationInfoOf<T>) -> Result<BalanceOf<T>, DispatchError> {
+		let slash_percentage = T::EvaluatorSlash::get();
+		let treasury_account = T::ProtocolGrowthTreasury::get();
+
+		// * Calculate variables *
+		// We need to make sure that the current PLMC bond is always >= than the slash amount.
+		let slashed_amount = slash_percentage * evaluation.original_plmc_bond;
+
+		T::NativeCurrency::transfer_on_hold(
+			&HoldReason::Evaluation(project_id).into(),
+			&evaluation.evaluator,
+			&treasury_account,
+			slashed_amount,
+			Precision::Exact,
+			Restriction::Free,
+			Fortitude::Force,
+		)?;
+
+		Ok(evaluation.current_plmc_bond.saturating_sub(slashed_amount))
+	}
+
+	fn reward_evaluator(
+		project_id: ProjectId,
+		evaluation: &EvaluationInfoOf<T>,
+		info: &RewardInfoOf<T>,
+	) -> Result<(BalanceOf<T>, BalanceOf<T>), DispatchError> {
+		let reward = Self::calculate_evaluator_reward(evaluation, &info);
+		Self::mint_ct_tokens(project_id, &evaluation.evaluator, reward)?;
```

Comment:
`mint_contribution_tokens_tokens`? lol
I would call it `mint_ct` or `mint_contribution_tokens`.

It's like saying send "xcm message" 😆